.github/workflows/test.yml

```
@@ -0,0 +1,41 @@
+name: Workspace Checks & Tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  check:
+    name: Clippy and fmt
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          components: rustfmt, clippy
+      - uses: Swatinem/rust-cache@v1
+      - name: Check Formatting
+        run: cargo fmt -- --check
+      - name: Check Clippy
+        run: cargo clippy -- -Dclippy::all
```

```suggestion
        run: cargo clippy --tests -- -Dclippy::all
```

Pretty optional but can help make tests more readable sometimes to make sure there aren't weird patterns followed